crates/types/src/node_id.rs

```diff
@@ -8,6 +8,15 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
+#[cfg(test)]
+use arc_swap::ArcSwap;
+use once_cell::sync::OnceCell;
+
+#[cfg(not(test))]
+static MY_NODE_ID: OnceCell<NodeId> = OnceCell::new();
+#[cfg(test)]
+static MY_NODE_ID: OnceCell<ArcSwap<NodeId>> = OnceCell::new();
```

Comment:
Building a MiniCluster based on multiple processes is probably the more powerful abstraction, e.g. when it comes to killing individual processes. So that is fine.

Another aspect, I want to add is that globals introduce implicit dependencies between components that might not be apparent when looking at the constructors. So testing one component can leave the globals in an incompatible state for another component which one needs to know. When working on Flink, I always liked to have things explicit (even if it cost a bit more in typing and plumbing) because it made the reasoning easier.

While the plumbing of information can be tedious at times, I never found it to be the limiting factor of productivity. For me, my speed of thinking was always the more constraining factor.